features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/index.jsx

```diff
@@ -806,7 +808,7 @@ class Details extends Component {
                                 </AccordianSummary>
                                 <AccordionDetails>
                                     <div>
-                                        {!api.isWebSocket() && (
+                                        {!api.advertiseInfo.advertised && !api.isWebSocket() && (
```

Comment:
have we handled when `advertiseInfo` is null here?

Reply:
Please check all the other places..

Please test this with normal non-advertise APIs as well?

Reply:
I have tested it with non advertise apis as well aiya. Worked fine with non advertise apis

Reply:
and also this should be an OR logic not and , Because runtime configs should be hidden when API is `advertised` **or** none-WebSocket APIs